pkg/logs/diagnostic/message_receiver.go

```diff
@@ -6,31 +6,29 @@
 package diagnostic
 
 import (
-	"context"
-	"fmt"
 	"sync"
-	"time"
 
 	"github.com/DataDog/datadog-agent/pkg/logs/config"
 	"github.com/DataDog/datadog-agent/pkg/logs/message"
-	"github.com/DataDog/datadog-agent/pkg/util/hostname"
 )
 
 // MessageReceiver interface to handle messages for diagnostics
 type MessageReceiver interface {
-	HandleMessage(message.Message, []byte)
+	HandleMessage(message.Message, string, []byte)
 }
 
 type messagePair struct {
 	msg         *message.Message
 	redactedMsg []byte
+	eventType   string
```

Comment:
Would it make sense to change this into an interface, with logs and event platform providing their own implementations (including formatting and filtering)?

Reply:
For the purposes of this comment, I am going to call this interface `FilterFormatter`, because naming is hard.

I am not sure to be honest. I like the idea a lot, but because the two implementations would need different metadata in order to handle their respective messages, it would mean the caller would have to create an instance of the `FilterFormatter` implementation and then call `HandleMessage` themselves, or they would have to define a helper function to do that themselves.

But `HandleMessage` calls `IsEnabled` to check whether it should create and enqueue the message. Which a`FilterFormatter` implementation would not have access to, so we would be creating objects just to drop them, which is not ideal. 

I can see that if the current implementation continues to be built on top of, it could become a jumbled mess. But I worry that trying to pull this portion of code out into its own interface, vs the `Formatter` interface that is introduced in this PR, would add a level of complexity that I am not sure would be worth it at this point.

I am happy to continue to go down this rabbit hole if you feel strongly about it however!